pkg/controller/kogitoapp/resource/manage_resources.go

```diff
@@ -0,0 +1,360 @@
+
+		update := false
+
+		if instance.Spec.Replicas != nil && *instance.Spec.Replicas != depConfig.Spec.Replicas {
+			log.Debug("Changes detected in 'Replicas'.", " OLD - ", depConfig.Spec.Replicas, " NEW - ", *instance.Spec.Replicas)
+			depConfig.Spec.Replicas = *instance.Spec.Replicas
+			update = true
+		}
+
+		resources := shared.FromResourcesToResourcesRequirements(instance.Spec.Resources)
+		if !reflect.DeepEqual(&depConfig.Spec.Template.Spec.Containers[0].Resources, resources) {
+			log.Debug("Changes detected in 'Resources'.", " OLD - ", depConfig.Spec.Template.Spec.Containers[0].Resources, " NEW - ", *resources)
+			depConfig.Spec.Template.Spec.Containers[0].Resources = *resources
+			update = true
+		}
+
+		if !reflect.DeepEqual(envToMap(instance.Spec.Env), envVarToMap(depConfig.Spec.Template.Spec.Containers[0].Env)) {
+			log.Debug("Changes detected in 'Env'.", " OLD - ", depConfig.Spec.Template.Spec.Containers[0].Env, " NEW - ", instance.Spec.Env)
+			depConfig.Spec.Template.Spec.Containers[0].Env = shared.FromEnvToEnvVar(instance.Spec.Env)
+			update = true
+		}
+
+		if update {
+			log.Info("Updating DeploymentConfig")
+
+			if err := kubernetes.ResourceC(client).Update(depConfig); err != nil {
+				return update, err
+			}
+
+			return update, nil
+		}
+	}
+
+	return false, nil
+}
+
+func ensureService(instance *v1alpha1.KogitoApp, service *corev1.Service, client *client.Client) (bool, error) {
+	if service != nil {
+		log := log.With("kind", service.GetObjectKind().GroupVersionKind().Kind, "name", service.Name, "namespace", service.Namespace)
+
+		if _, err := kubernetes.ResourceC(client).Fetch(service); err != nil {
+			return false, err
+		}
+
+		if update := ensureServiceLabels(instance, service.Labels, log); update {
+			log.Info("Updating Service")
+
+			if err := kubernetes.ResourceC(client).Update(service); err != nil {
+				return update, err
+			}
+
+			return update, nil
+		}
+	}
+
+	return false, nil
+}
+
+func ensureRoute(instance *v1alpha1.KogitoApp, route *routev1.Route, client *client.Client) (bool, error) {
+	if route != nil {
+		log := log.With("kind", route.GetObjectKind().GroupVersionKind().Kind, "name", route.Name, "namespace", route.Namespace)
+
+		if _, err := kubernetes.ResourceC(client).Fetch(route); err != nil {
+			return false, err
+		}
+
+		if update := ensureServiceLabels(instance, route.Labels, log); update {
+			log.Info("Updating Route")
+
+			if err := kubernetes.ResourceC(client).Update(route); err != nil {
+				return update, err
+			}
+
+			return update, nil
+		}
+	}
+
+	return false, nil
+}
+
+func ensureServiceLabels(instance *v1alpha1.KogitoApp, serviceLabels map[string]string, log *zap.SugaredLogger) bool {
+	update := false
+
+	labels := map[string]string{}
+	labels[LabelKeyAppName] = instance.Name
+	if len(instance.Spec.Service.Labels) > 0 {
+
+		for key, value := range instance.Spec.Service.Labels {
+			labels[key] = value
+		}
+	} else {
+		labels[LabelKeyServiceName] = instance.Name
```

Comment:
This is the default label, right? WDYT having a func to add default service labels and then we reuse it here and when creating the service?

Reply:
moved the function to [meta.go](https://github.com/kiegroup/kogito-cloud-operator/blob/33e9918d0604a0593a609031c21facc071cb286f/pkg/controller/kogitoapp/resource/meta.go#L69)